core/src/main/java/fr/sncf/osrd/standalone_sim/SafetySpeed.kt

```diff
@@ -79,6 +113,9 @@ fun getSignalOffsets(
         }
         prevZonePathsLength += rawInfra.getZonePathLength(zonePath).distance
     }
+    // Add one "signal" at the end of the last route no matter what.
+    // There must be either a signal or a buffer stop, on which we may end safety speed ranges.
+    res.add(Offset(prevZonePathsLength - pathStartOffset.distance))
```

Comment:
Is it possible that there is a signal attached to the bufferStop (or the routeExit) and we are "hiding" it if choosing this new one?

Reply:
In the current version, the default safety speed applied at the buffer stop will always end at the buffer stop.

But if there's a signal a few meters before, it would still be included in the `signalOffsets`. It would be used if there's a stop on closed signal before that (on top of the buffer stop safety range)

Reply:
As discussed, I resolve this:
* there could be a wrong offset on the safety-speed range if we ever have explicit signal upstream a buffer-stop, but it is not the case for France
* there could be an issue if the train finishes between the final signal of the route and the final detector of the route, but it's an "undefined behavior" case
* As for the final route not being the "best one" for the simulation, the usual issues are here:
  * we could apply a safety-speed because of a buffer-stop that is after the end of the path and choosing another route we would have avoided it
  * we could miss the application of a buffer-stop's safety-speed just because it's on a very short route after the last one used on the path